api/proto/teleport/usageevents/v1/usageevents.proto

```diff
@@ -444,6 +444,41 @@ message DiscoveredDatabaseMetadata {
   string db_protocol = 2;
 }
 
+// FeatureRecommendationEvent captures event emitted when a feature is recommended to user or
+// when user completes the desired CTA for the feature.
+//
+// PostHost event: tp.ui.feature.recommendation
+message FeatureRecommendationEvent {
+  // anonymized Teleport username, 32 bytes (HMAC-SHA-256) encoded in base64;
+  //
+  // PostHog property: tp.user_name
+  string user_name = 1;
+  // feature is name of the Teleport feature.
+  //
+  // PostHost property: tp.feature
+  Feature feature = 2;
+  // feature_recommendation_status records recommendation state, which can be 'NOTIFIED' (feature is recommended)
+  // OR 'DONE' (user completes desired CTA)
+  //
+  // PostHost property: tp.feature_recommendation_status
+  FeatureRecommendationStatus feature_recommendation_status = 3;
+}
+
+// Feature is name of Teleport feature
+enum Feature {
+  FEATURE_UNSPECIFIED = 0;
+  FEATURE_TRUSTED_DEVICES = 1;
+}
+
+// FeatureRecommendationStatus is feature recommendation status.
+enum FeatureRecommendationStatus {
+  FEATURE_RECOMMENDATION_STATUS_UNSPECIFIED = 0;
+  // FEATURE_RECOMMENDATION_STATUS_NOTIFIED is emitted when a feature is recommended (notified in UI) to user.
+  FEATURE_RECOMMENDATION_STATUS_NOTIFIED = 1;
+  // FEATURE_RECOMMENDATION_STATUS_DONE is emitted when user completes the desired CTA.
+  FEATURE_RECOMMENDATION_STATUS_DONE = 2;
```

Comment:
Just to a small question to be safe. In the Cloud PR, you wrote that you want to measure "time between "red dot" shown in the UI and device enrollment". I assume that this would be the time between either status notified and enrolling a device or status notified and status done.

Are you positive that sending an event with status done won't overwrite the one sent earlier in status notified?

I'm not an expert on Posthog, but I remember that some properties work this way. Though I think that happens when the property is set on the user itself and not on an event.

Reply:
> Are you positive that sending an event with status done won't overwrite the one sent earlier in status notified?

Yes I am positive, I have tested this and it does not override but creates a new event for that (I have Prehog dev setup locally and test with it before the Cloud PR, then test it again once Cloud PR is merged). 

Thanks for pointing out that though, looks like something to think about. I did a quick search and results only came up for Posthog feature flags. (time to up my 🔍 skills!)